src/modm/math/filter/moving_average.hpp

```diff
@@ -82,7 +82,7 @@ class MovingAverage
 	{
 		if constexpr(std::floating_point<T>) {
 			buffer[index] = input;
-			sum = std::accumulate(std::begin(buffer), std::end(buffer), 0);
+			sum = std::accumulate(std::begin(buffer), std::end(buffer), 0.0);
```

Comment:
```suggestion
			sum = std::accumulate(std::begin(buffer), std::end(buffer), 0.0f);
```
Perhaps even this to prevent double deduction?

Reply:
```suggestion
			sum = std::accumulate(std::begin(buffer), std::end(buffer), T{0});
```

On some targets modm compiles with `-fsingle-precision-constant`. That's why 0.0 is interpreted as a `float` constant instead of `double`. On other platforms with hardware 64-bit float support that behaviour is not enabled and the type would be deduced as `double` with `0.0`. `T{0}` should do the right thing in all circumstances.

Reply:
Hi @Artiom9,

you have marked the conversation as resolved but did not push the review suggestion. Could you do that or should I? Afterwards we would merge the PR.

Reply:
Hi, @chris-durand! To be honest, I do not really understand the Githab and do not really understand the procedure of the pull request. I just wanted to point out the problem and provide my solution. If some other actions are needed, do it for me. Thank you.

Reply:
Thank you very much for contributing the fix! I have squashed the commits and we will merge it soon.